src/coap_client_libcoap.c

```diff
@@ -499,6 +535,43 @@ static void golioth_coap_post(golioth_coap_request_msg_t *req, coap_session_t *s
     coap_send(session, req_pdu);
 }
 
+static void golioth_coap_post_block(golioth_coap_request_msg_t *req,
+                                    struct golioth_client *client,
+                                    coap_session_t *session)
+{
+    coap_pdu_t *req_pdu = coap_new_pdu(COAP_MESSAGE_CON, COAP_REQUEST_CODE_POST, session);
+    if (!req_pdu)
+    {
+        GLTH_LOGE(TAG, "coap_new_pdu() post failed");
+        return;
+    }
+
+    if (req->post_block.block_index == 0)
+    {
+        // Save this token for further blocks
+        golioth_coap_add_token(req_pdu, req, session);
+        memcpy(client->block_token, req->token, req->token_len);
```

Comment:
Is `client->block_token` also used for block downloads? Seems like this could get clobbered if we try to do more than one blockwise operation at a time.

If the answer is yes, let's file a ticket for allowing multiple blockwise transfers at the same time. No need to address in this PR.

Reply:
Yes, it is used by both. Here's the new issue:
https://github.com/golioth/firmware-issue-tracker/issues/569